modules/blocks.js

```diff
@@ -240,14 +241,15 @@ Blocks.prototype.onBind = function(scope) {
  * @todo Add @returns tag
  */
 Blocks.prototype.onNewBlock = function(block) {
-	const tasks = [
-		modules.cache.KEYS.blocksApi,
-		modules.cache.KEYS.transactionsApi,
-	].map(pattern => callback => modules.cache.clearCacheFor(pattern, callback));
+	if (components.cache) {
+		const tasks = [CACHE.KEYS.blocksApi, CACHE.KEYS.transactionsApi].map(
+			pattern => callback => components.cache.removeByPattern(pattern, callback)
+		);
 
-	async.parallel(async.reflectAll(tasks), () =>
-		library.network.io.sockets.emit('blocks/change', block)
-	);
+		async.parallel(async.reflectAll(tasks), () =>
```

Comment:
It seem previously this was always being called now it seems it's tied to `components.cache` to be truthy else is not going to be called. If `components.cache` is always true then we don't need the if or maybe this change should keep the `async.parallel...` call outside the if scope?

Reply:
Good catch.